docs/groovy/how-to-guides/date-time-literals.md

```diff
@@ -0,0 +1,138 @@
+---
+title: Date-time literals in query strings
+sidebar_label: Date-time
+---
+
+Date-time literals in query strings directly correspond to data types from the [java.time](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/time/package-summary.html) package. Deephaven leverages these Java date-time types to perform date-time arithmetic and comparisons within queries, allowing you to work with dates, times, durations, periods, and more using familiar and precise formats.
+
+Proper handling of date-time values in queries is critical in Deephaven. For complete coverage of this topic, see [Work with time in Deephaven](../conceptual/time-in-deephaven.md).
+
+This guide provides comprehensive coverage of date-time literals in query strings.
+
+## What denotes a date-time literal?
+
+In query strings, date-time literals are denoted by single quotes (`'`) around a string that represents a date-time data type such as a moment in time, duration, period, etc. Consider the following code:
+
+```groovy order=source,sourceMeta
+source = emptyTable(1).update(
+    "InstantLiteral = '2025-05-01T00:00:00Z'",
+    "DurationLiteral = 'PT5H2M1.2S'",
+    "InstantPlusDuration = InstantLiteral + DurationLiteral",
+)
+sourceMeta = source.meta()
+```
+
+Literal values are converted into date-time literals. If the conversion is successful, the value is treated as a date-time literal in the [formula](./formulas.md). If the value is improperly formatted, the conversion will fail and an error will be raised.
+
+Deephaven highly recommends the use of double quotes (`"`) for query strings because single quotes (`'`) denote date-time and character literals. If you do use single quotes to enclose query strings, you must escape any enclosed single quotes, as shown by the example below:
+
+```groovy order=source
+// The query string below is enclosed in single quotes
+// As a result, the inner single quotes must be escaped
+// This approach is less readable
+source = emptyTable(1).update('InstantLiteral = \'2025-05-01T00:00:00Z\'')
```

Comment:
So I noticed that, in the Python guide, I actually used double quotes for this example. Can you fix that to look like this? That one's on me.